pkg/apis/flow/v1alpha1/common_types.go

```diff
@@ -18,20 +18,55 @@ package v1alpha1
 
 import (
 	corev1 "k8s.io/api/core/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
 )
 
+/* Provide common structs that are used by the targets such as secret definitions */
+
 // ValueFromField is a struct field that can have its value either defined
 // explicitly or sourced from another entity.
 type ValueFromField struct {
 	// Optional: no more than one of the following may be specified.
 
 	// Field value.
 	// +optional
-	Value *string `json:"value,omitempty"`
+	Value string `json:"value,omitempty"`
```

Comment:
The rationale is the same as in Kubernetes' `corev1.EnvVar` (which was my inspiration for creating this type originally): it is enough to check for `""` because an empty string will be omitted anyway while (de-)serializing to/from JSON.

With that being said, the removal of the pointer type here is only a result of me copying the type from `sources` / `routing` / `target` for consistency. I'm totally fine reverting to a pointer if we agree that this becomes a pointer in other component types too.

Reply:
I'm totally game with keeping this as a generic string and do get your point on the serialization/deserializtion. Upon first glance, I thought this was introducing an inconsistency instead of correcting for it.